contrib_bots/lib/xkcd.py

```diff
@@ -0,0 +1,120 @@
+import requests
+import logging
+from random import randint
+
+class xkcdHandler(object):
+    '''
+    This plugin provides several command that can be used for fetch comics from
+    xkcd. The bot looks for messages starting with "@xkcd" and responds with a
+    message with the comic based on provided commands.
+    '''
+
+    XKCD_TEMPLATE = 'http://xkcd.com/%s/info.0.json'
+    LATEST_XKCD = 'http://xkcd.com/info.0.json'
+
+    def usage(self):
+        return '''
+            This plugin allows users to fetch comics provided by xkcd.
+            Users should preface the command with "@xkcd".
+
+            There are several commands to use this bot:
+            - @xkcd help -> To show all commands the bot supports.
+            - @xkcd latest -> To fetch latest comic from xkcd.
+            - @xkcd random -> To fetch random comic from xkcd.
+            - @xkcd <comic_id> -> To fetch comic based on `<comic_id>`, e.g
+              `@xkcd 1234`.
+            '''
+
+    def fetch_xkcd_query(self, mode, comic_id=None):
+        try:
+            if mode == 0:  # Fetch latest comic.
+                url = self.LATEST_XKCD
+
+            elif mode == 1:  # Fetch random comic.
+                latest = requests.get(self.LATEST_XKCD)
+                latest_id = latest.json()['num']
+                randommed = randint(1, latest_id)
```

Comment:
Could you have a better variable name here? I don't think `randommed` is a word.